node/src/main/scala/org/bitcoins/node/NodeCallbacks.scala

```diff
@@ -1,133 +1,164 @@
+    extends Callback[Vector[(DoubleSha256Digest, GolombFilter)]]
+
+/** Callback for handling a received block header */
+trait OnBlockHeadersReceived extends Callback[Vector[BlockHeader]]
+
 object NodeCallbacks {
 
+  // Use Impl pattern here to enforce the correct names on the CallbackHandlers
+  private case class NodeCallbacksImpl(
+      onCompactFiltersReceived: CallbackHandler[
+        Vector[(DoubleSha256Digest, GolombFilter)],
+        OnCompactFiltersReceived],
+      onTxReceived: CallbackHandler[Transaction, OnTxReceived],
+      onBlockReceived: CallbackHandler[Block, OnBlockReceived],
+      onMerkleBlockReceived: CallbackHandler[
+        (MerkleBlock, Vector[Transaction]),
+        OnMerkleBlockReceived],
+      onBlockHeadersReceived: CallbackHandler[
+        Vector[BlockHeader],
+        OnBlockHeadersReceived]
+  ) extends NodeCallbacks {
+
+    override def +(other: NodeCallbacks): NodeCallbacks =
+      copy(
+        onCompactFiltersReceived = onCompactFiltersReceived ++ other.onCompactFiltersReceived,
+        onTxReceived = onTxReceived ++ other.onTxReceived,
+        onBlockReceived = onBlockReceived ++ other.onBlockReceived,
+        onMerkleBlockReceived = onMerkleBlockReceived ++ other.onMerkleBlockReceived,
+        onBlockHeadersReceived = onBlockHeadersReceived ++ other.onBlockHeadersReceived
+      )
+  }
+
   /** Constructs a set of callbacks that only acts on TX received */
   def onTxReceived(f: OnTxReceived): NodeCallbacks =
-    NodeCallbacks(onTxReceived = Seq(f))
+    NodeCallbacks(onTxReceived = Vector(f))
 
   /** Constructs a set of callbacks that only acts on block received */
   def onBlockReceived(f: OnBlockReceived): NodeCallbacks =
-    NodeCallbacks(onBlockReceived = Seq(f))
+    NodeCallbacks(onBlockReceived = Vector(f))
 
   /** Constructs a set of callbacks that only acts on merkle block received */
   def onMerkleBlockReceived(f: OnMerkleBlockReceived): NodeCallbacks =
-    NodeCallbacks(onMerkleBlockReceived = Seq(f))
+    NodeCallbacks(onMerkleBlockReceived = Vector(f))
 
   /** Constructs a set of callbacks that only acts on compact filter received */
   def onCompactFilterReceived(f: OnCompactFiltersReceived): NodeCallbacks =
-    NodeCallbacks(onCompactFiltersReceived = Seq(f))
+    NodeCallbacks(onCompactFiltersReceived = Vector(f))
 
   /** Constructs a set of callbacks that only acts on block headers received */
   def onBlockHeadersReceived(f: OnBlockHeadersReceived): NodeCallbacks =
-    NodeCallbacks(onBlockHeadersReceived = Seq(f))
+    NodeCallbacks(onBlockHeadersReceived = Vector(f))
 
   /** Empty callbacks that does nothing with the received data */
   val empty: NodeCallbacks =
-    NodeCallbacks(
-      onTxReceived = Seq.empty,
-      onBlockReceived = Seq.empty,
-      onMerkleBlockReceived = Seq.empty,
-      onCompactFiltersReceived = Seq.empty,
-      onBlockHeadersReceived = Seq.empty
+    apply(Vector.empty, Vector.empty, Vector.empty, Vector.empty, Vector.empty)
```

Comment:
Let's not use `apply()` like this as it is super confusing. Just use the proper name `NodeCallbacks()`